### ISSUE TITLE
VM Image Template of PMSS, to be easily used with proxmox

### ISSUE THREAD

**Issue by MagnaCapax** [MagnaCapax/PMSS#53]
We probably should start enrolling a VM Image template (or perhaps Deb10-Deb12 templates) for PMSS upon releases.

Need to automate it significantly, dev scripts to roll up a VM image into releaseable image, and push to our static server as well.

Should clear out all ID type of things, ssh keys, etc. (see mcxSauces cloner script).
Companion script for proxmox install new VM with PMSS, arguments should be IP address to use, VMID. We already have some internal scripts to do this, we'll look them up and probably use as a basis for release (releasing to public needs additional QA/QC for different types of use, edge cases etc.)

We can then ourselves use that as well when setting up new guests as seedboxes, dev environments etc.

**Comment by MagnaCapax** [comment]
## Investigation Report

**Issue:** #53 — VM Image Template of PMSS, to be easily used with proxmox
**Investigated:** 2026-06-01
**Status:** PARTIAL

### Summary
The PMSS-side prerequisite for this issue is already done: `install.sh` is fully unattended-capable. The generalize/sysprep and image-assembly machinery this issue describes is also already built — but it lives in PM's dedicated image-templating tooling, *not* in the PMSS repository, and that is the correct home for it. What remains (baking PMSS into an image, a Proxmox provisioning wrapper, and a public-QA release) is image-pipeline/dev-ops work that belongs in that tooling, not in PMSS source. That is also why automated build attempts here produced no commits: there is nothing for this issue to change inside the PMSS repo.

### What The Issue Asks For
1. Dev scripts that roll a VM into a releasable PMSS image and publish it.
2. Generalize/sysprep the image (clear machine-id, SSH host keys, etc.).
3. A companion Proxmox "create new VM with PMSS" script taking an IP and a VMID.
4. A public release, gated on additional QA/QC (per the issue text itself).

### Implementation Status
- **Unattended PMSS install — DONE.** `install.sh` supports `--hostname=`, `--quota-mount=`, `--non-interactive`, `--skip-hostname`, `--skip-quota`, `--skip-upgrade`, `--dry-run`, `--skip-update`. `docs/install.md` and ADR 0007 document and codify the unattended contract. So the "install PMSS into the image" step needs no new install machinery.
- **Generalize/sysprep — DONE (and exceeds the ask), but outside PMSS.** PM's dedicated image-templating tooling already resets machine-id and SSH host keys, clears cloud state, and scrubs logs/caches/udev/entropy, then renders hostname/network/fstab. It is multi-distro and CI-tested, and it exposes a checksum-verified post-install hook — the natural plug point for `install.sh`.
- **Image build pipeline — DONE generically, outside PMSS.** That same tooling builds image artifacts from a live host or a disk image and validates them.
- **PMSS source — nothing here.** `scripts/` is per-user management plus `update.php`; there is no template/sysprep/Proxmox/clone code and no such history. The repo `Dockerfile` is validation-only.

### What's Missing
- Wiring PMSS install into the existing checksum-verified post-install hook (small glue, belongs in the tooling).
- A thin Proxmox provisioning wrapper (`qm create` + disk import + IP/VMID). No such script exists anywhere today; VM creation is currently manual or "cloud image + install.sh".
- The public-QA release (the issue defers this explicitly).

The internal cloner the issue references targets **bare-metal** nodes (USB-boot + large-NVMe BIOS workaround), not Proxmox VMs — so it is not a basis for the Proxmox half.

### Mission Alignment
Serves value-for-money (#5: clean one-shot deploys avoid per-deploy install time and the documented stale-template clone-debt), reliability (#7: reproducible images eliminate config drift), and sovereignty (#6: PM's own image pipeline). No privacy/customer-data exposure — this is infra-side provisioning. Honesty (#2/#3): the report does not claim the issue is done; it states precisely which half exists and where the rest belongs.

### Devil's Advocate
*Why not build it here?* Because PMSS is seedbox server-side code; image-build/release orchestration is dev-ops tooling, and PM already maintains it separately. Duplicating it in PMSS would violate separation of concerns and ADR 0007 (which keeps `install.sh` a thin bootstrapper). *Rebuttal that it should still be built:* yes — but in the dedicated tooling, where the engine and the post-install hook already are; the residual work is glue plus a small Proxmox wrapper, not a new build system.

### Anti-Pattern Hunt
- **REGEX_FOR_SEMANTICS:** PASS — no semantic enforcement via pattern-matching is proposed.
- **PREMATURE_OPTIMIZATION:** PASS — the image-build engine already Exists, is Tested (CI), and is Useful (drives provisioning); rebuilding it in PMSS is the premature/duplicate path, which this report rejects.
- **FALSE_COMPLETION_CLAIM:** PASS — status is PARTIAL with an explicit done/missing split; no "complete" claim.
- **OPERATOR_HANDOFF_ANTIPATTERN:** PASS — a decided recommendation with a concrete re-home path; the issue is left open only for operator visibility because a cross-repo close is not auto-closable here.
- **FABRICATED_AUTHORITY_OR_PATH:** PASS — every cited flag/ADR/file was read this session (`install.sh`, `docs/adr/0007`, `README.md`, `docs/install.md`).
- **FAIL_OPEN_SAFETY_BEHAVIOR:** PASS — the post-install hook fails closed (checksum mismatch aborts; download failure skips), and cloud-init must be purged after provisioning rather than left resident.
- **DRY_RUN_THEATER:** PASS — N/A; `install.sh --dry-run` is a real parse-and-print path (verified), not claimed as proof of anything.
- **INVESTIGATION_LOOP_THEATER:** PASS — prior attempts produced no verdict; this one reaches a decision (don't build here; re-home) and breaks the loop.

### Related Incidents & Data
- A stale Debian-10 VM template once caused mail-spool **inode exhaustion across 6 cloned VMs** plus legacy-script breakage (~1.5h debugging). That incident is the strongest argument *for* a clean baked image — and against clone-from-old-template. (Operator has since reaffirmed wanting a baked release image.)
- This issue has been picked up by automated build cycles multiple times and produced **zero commits each time** — consistent with the issue being filed against the wrong repository, not with it being too complex.

### Cross-Issue Analysis
- **#111 (Debian 13 validation roadmap):** supplies the distro/version targets an image pipeline consumes — a soft dependency, not an overlap.
- **#51 (automatic benchmarks after install):** the "QA/QC" gate this issue mentions could reuse #51's post-provision QC.
- **#171 (container runtime options for user isolation):** an alternative isolation model (container vs VM) — adjacent, not overlapping.
No direct duplicate exists in the open issue set.

### Dependencies & Blockers
- The remaining work depends on the dedicated image-build tooling (separate repo), not on PMSS changes.
- Distro/version targets are informed by #111.
- No research gap — internal evidence is sufficient to decide.

### Reuse, Refactor & Test Strategy
- **Existing reuse points:** `install.sh` unattended flags; the dedicated tooling's image-assembly, sysprep, and checksum-verified post-install hook; the existing documented "cloud image / netinstall + `install.sh`" recipe.
- **Refactor opportunities:** install PMSS by pointing the existing post-install hook at `install.sh` — no new code in PMSS. Do **not** fatten `install.sh` (ADR 0007).
- **Test categories proposed (for the tooling repo):** hermetic unit (template assembly); integration (provision a VM on a dev Proxmox host → run `install.sh` → `scripts/util/systemTest.php`); regression (the Debian-10 clone inode/exim debt); boundary (IP/VMID argument validation); failure injection (corrupt post-install checksum → hook must abort).
- **Minimum-code-for-correctness:** extract template → post-install-hook runs `install.sh --non-interactive --hostname=<fqdn>` → thin `qm` wrapper. No new abstractions; no speculative configurability.
- **Mission/business success signal:** time-to-deploy a clean seedbox VM drops from a manual multi-step process to one command (mission #5, efficiency); business signal = faster capacity turn-up and zero recurrence of clone-debt incidents (ops-cost reduction).

### Deep Research Gate
SKIPPED: internal-evidence-sufficient — the relevant tooling, install contract, and operator architecture decisions were all locatable and read directly; this issue has also been investigated multiple times already.

### Architectural Minimization
- **Step A — dumbest requirement:** "push to *our static server*" bakes in a bespoke 2024 distribution channel. WHO: the 2024 issue author, predating the later date-coded VM-per-distro image decision. The real constraint is "a fetchable, versioned image artifact"; the specific static-server mechanism is an inherited assumption that should defer to the current versioned image pipeline.
- **Step B — alternatives:**
  - **DELETE:** (1) Drop the custom image entirely and use "cloud image / netinstall + unattended `install.sh`" (already exists) — rejected as a *full* substitute because the operator explicitly wants a baked release image for instant, variance-free deploys. (2) Delete any new "build a PMSS image" code by reusing the existing assemble + post-install hook — recommended.
  - **MERGE:** (1) Merge this issue's image-build into the existing dedicated templating engine (already does assemble + sysprep + post-install hook) — **PREFERRED**. (2) Merge the "Proxmox provision" half by promoting the existing manual VM-creation recipe into a thin scripted wrapper — candidate.
  - **REFACTOR:** (1) Add a PMSS flavor to the existing post-install hook usage. (2) Generalize `install.sh` to do orchestration — rejected (ADR 0007 forbids fattening it).
  - **NULL:** If we do nothing, deploys stay manual and the stale-template clone-debt failure mode can recur (the documented inode-exhaustion incident).
  - **Winner:** the MERGE-into-dedicated-tooling alternative wins.
- **Step C:** Alternative wins — no minimal design is proposed *for PMSS*, because the work merges into the dedicated image-build tooling, not the PMSS source tree.

### ADR Adherence
ADHERES: ADR 0007 (Install bootstrap interactivity and contract) — the recommendation explicitly keeps image-build/Proxmox orchestration out of `install.sh`, preserving the thin-bootstrapper contract. Reviewed all 18 ADRs (0003–0020); none others address image/provisioning, so no further ADR is relevant.

### Final AQFM Sweep
HELD — a memory synthesis over the full draft confirmed the dedicated tooling is the image-build engine, that the operator decided on a VM-per-distro image architecture, and that PM's own images are an active (not abandoned) goal. None of this revises the recommendation; it reinforces that #53 is the PMSS-Debian-Proxmox slice of a broader effort whose home is dedicated tooling.

### Adversarial Self-Challenge
**Strongest finding 1:** No automated Proxmox VM provisioning script exists anywhere.
- Disproving search 1: broad search for `qm create`/`pct create`/`qm clone`/`pvesh.*create` across PMSS, the dedicated tooling, and infra tools → only management/monitoring tooling (zombie-VM remediation, blast-radius classification, capacity collection); none create a VM.
- Disproving search 2 (broader): provisioning lessons in memory → VM creation is documented as **manual** ("copy qcow2 boot disk + write VM config directly").
- Outcome: held.

**Strongest finding 2:** The dedicated tooling already implements #53's generalize/sysprep half.
- Disproving search: scan the sysprep tasks for `TODO`/`FIXME`/`stub`/`not implemented` → none found; tasks are concrete and CI-covered.
- Outcome: held.

**Least-evidenced finding** (lowest citation count): The remaining work belongs in dedicated tooling rather than the PMSS public repo (an architectural inference).
- Disproving search 1: memory for operator intent on image-build location/open-sourcing → the documented operator decision places image building on a dedicated dev host with a per-distro pipeline, not in PMSS source.
- Disproving search 2 (broader): bare-metal provisioning core research → PM's provisioning consistently runs through dedicated systems, not the PMSS application repo.
- Outcome: held.

### Recommendation
**SHOULD-NOT-IMPLEMENT** (in this PMSS repository).
The PMSS-side prerequisite (unattended `install.sh`) is already complete and is correctly bounded by ADR 0007. The image-build/sysprep engine already exists in PM's dedicated image-templating tooling, which is the right home for it. The remaining work — wiring PMSS install into the existing post-install hook, a thin Proxmox provisioning wrapper, and the public-QA release — is image-pipeline/dev-ops work that should be tracked and built in that tooling, not added to PMSS source. The need is real and operator-endorsed; only its location is being corrected.

If PM wants self-hosters to benefit publicly, a *small, secret-free* "provision a PMSS VM on Proxmox" helper or documented recipe could legitimately live here later — that is an operator call, separate from the internal release pipeline.

### Auto-Close Decision
AUTO_CLOSED_BLOCKED: trigger-not-met; reason=rationale-not-supersession — the SHOULD-NOT-IMPLEMENT verdict is for separation-of-concerns / re-homing (ADR 0007), not supersession by an open issue in this repo; there is no open successor issue in this repository to cite, so the issue is left open for operator visibility on the re-home decision.

### Additive Recommendation Flags
- **RECOMMEND_NEW_ISSUE** — track the residual work in PM's dedicated image-build tooling. Draft title: *"PMSS release image: wire install.sh into the post-config hook + thin Proxmox (qm) provisioning wrapper (IP/VMID)"*. Scope: one baked, sysprepped PMSS release image per supported Debian target, plus a minimal Proxmox create-and-configure wrapper, reusing the existing assemble/sysprep/post-install machinery.

— Väinämöinen

**Comment by MagnaCapax** [comment]
## Adversarial Review

**Issue:** #53 — VM Image Template of PMSS, to be easily used with proxmox
**Reviewed:** 2026-06-10
**Verdict:** SHOULD-NOT-IMPLEMENT *(scoped to this repository)*

A prior investigation report (2026-06-01) reached the same conclusion. This pass independently re-verified every load-bearing claim against current source rather than trusting that report, sharpens the architecture argument, and — critically — applies the machine-readable verdict label the prior pass left off, which is why the autonomous build cycle kept re-selecting this issue.

### Adversarial Findings

**The Big Questions**

1. **Worst case (if built here as described):** image-bake + Proxmox `qm` provisioning logic lands in this repo → `install.sh` is fattened past its thin-bootstrapper contract (ADR 0007), *or* a hypervisor-host tool lands in a guest-overlay repo → the single explicit install path gains a special case → break-glass installs (which assume `install.sh` may be the only file present) inherit orchestration they cannot satisfy → fragile installs on exactly the recovery path that must stay simple. Blast radius is fleet-wide because `install.sh` runs on every install.
2. **Inversion (guarantee maximum damage):** put hypervisor provisioning + image distribution inside the bootstrapper, couple it to the install contract, and ship it publicly with internal network/boot/IP assumptions baked in. The proposed "add a companion script + roll images here" is closer to that than to a clean design — not because the intent is wrong, but because the *location* is.
3. **Regret in 6 months:** a `qm`-wrapper in this repo would need per-Debian-release retest, `qm` API-drift tracking, and image-checksum upkeep — recurring infra maintenance misfiled in a code repo, drifting from the real internal pipeline.
4. **Removal inversion:** if a PMSS-resident image pipeline already existed and someone proposed removing it to the separate dev-ops tooling, we would not fight to keep it here — which means we should not add it here.
5. **Dumbest requirement:** "push to *our static server*" — a bespoke 2024-era distribution channel baked into the ask. The real requirement is "a fetchable, versioned image artifact"; the static-server mechanism is an inherited assumption that should defer to the current versioned image pipeline.
6. **Deletion opportunity:** the function is achievable with *zero* new pipeline code here — reuse the already-unattended `install.sh` from the separate tooling's post-config hook, and the already-documented cloud-image/netinstall + `install.sh --non-interactive` recipe. The best part is no part.

**Finding items**

- [PREMATURE_OPTIMIZATION — rejected-as-duplicate] The generic image-build + sysprep engine already exists, is CI-tested, and is in use in PM's separate dev-ops tooling. Re-building it in this repo is the duplicate path; this review rejects it. (Engine Exists/Tested/Useful is satisfied — *elsewhere*.)
- [SCOPE-CONFLATION] The issue bundles four deliverables across two repos: (1) image build+publish, (2) sysprep/generalize, (3) Proxmox provisioning wrapper, (4) public release. (1)+(2) already exist in separate tooling; (4) is explicitly deferred by the issue itself ("needs additional QA/QC"); only (3) is genuinely missing anywhere — and it belongs on the hypervisor, not here.
- [LAYER-MISMATCH] A Proxmox provisioning wrapper runs `qm create`/`qm importdisk`/`qm set` on the **hypervisor host** to stand up a VM. PMSS is the overlay that runs **inside** the resulting guest. These are different architectural layers. A hypervisor-provisioning tool has no more business in this seedbox-overlay repo than a switch-config script does. (This sharpens the prior report, which hedged that a thin wrapper "could live here later" — it should not.)
- [MISLABEL — contributor trap] `good first issue` is actively misleading. This is a cross-repo, image-pipeline + hypervisor-provisioning + public-QA effort that defeated four automated build attempts and required deep investigation to conclude it does not belong here. A newcomer who picks it up will burn hours discovering it is mis-located. Recommend the operator remove `good first issue`.
- [PROCESS-SMELL — loop enabler] The 2026-06-01 investigation reached SHOULD-NOT-IMPLEMENT but applied no label, leaving the issue carrying only its original triage labels (`good first issue` among them) and no machine-readable verdict. With nothing to signal "stop," the autonomous build cycle has no reason not to re-select it (two BLOCKED 0-commit comments already on record). This pass applies `should-not-implement` to break the loop.

### Caller Analysis

This issue **adds** capability; it removes/replaces no existing function, so the removal HARD GATE and Chesterton/caller-enumeration trigger do not fire. The one adjacent risk worth stating for the implementer: any attempt to satisfy this by adding orchestration to `install.sh` would change the behavior of its **default invocation** — a compatibility contract relied on by the documented piped one-liner bootstrap installer, by unattended cron/CI runs, and by the break-glass recovery path (ADR 0007 notes `install.sh` may be the only file available). That is a fleet-wide contract surface; do not touch it for this feature.

### Investigation Results

Verified directly this session (not inherited from the prior report):

- **`install.sh:127-219, 197-208`** — unattended contract is complete: `--hostname=`, `--skip-hostname`, `--quota-mount=`, `--skip-quota`, `--non-interactive`, `--skip-upgrade`, `--dry-run`, `--skip-update`. `--dry-run` is a real parse-and-exit path (`install.sh:519-521`). The PMSS-side prerequisite of this issue is already shipped.
- **`docs/adr/0007-install-bootstrap-interactivity-and-contract.md`** — codifies `install.sh` as a thin bootstrapper: "heavyweight orchestration belongs in `update.php` and `update-step2.php`," with bootstrap prompts "intentionally limited." Image-bake/Proxmox orchestration must not be added here.
- **No VM-templating code or history** — `grep` across `scripts/`, `install.sh`, `etc/`, `util/` for sysprep/machine-id/proxmox/`qm`/clone returns only one unrelated rutorrent regex. `git log -S "qm create"` and `git log -S "machine-id"` are both empty. `Dockerfile` is validation-only.
- **`docs/install.md` (49 lines)** — documents the unattended flags (lines 8, 17) but contains **no** "provision a PMSS VM (cloud-image/netinstall + `install.sh --non-interactive`)" recipe. This is the single genuine, repo-appropriate gap (see Recommendations).
- **Proxmox creation is manual** — verified the workflow is a hand-run `qm create` + `qm importdisk` (does not auto-attach) + `qm set --scsi0/--boot/--ipconfig0` sequence; no automated wrapper exists. Cloud-init, if used, must be purged post-provision (known attack vector) — any wrapper must account for this.
- **No duplicate; cross-issue confirmed open:** #111 (Debian 13 roadmap — supplies distro targets, soft dependency), #51 (post-install benchmarks/QC — could supply the QA gate), #171 (container-vs-VM isolation — adjacent). No open duplicate in the issue set.

evidence mined: searched all-state issues "VM image qcow2 provisioning" / "image template proxmox VM" -> surfaced #53,#111,#170,#51,#171 -> used #111/#51/#171 (cross-issue) + confirmed no duplicate.

### Recommendations

- **Do not implement the pipeline in this repo.** The PMSS slice (unattended install) is done and correctly bounded by ADR 0007; the build/sysprep engine already exists in separate dev-ops tooling, which is the right home for deliverables 1–3. Track the residual (wire install into the post-config hook + a thin Proxmox `qm` wrapper + public QA) there, not here.
- **The one legitimately PMSS-appropriate, public-safe deliverable is docs-only:** a short, secret-free recipe in `docs/install.md` — "provision a PMSS VM: boot a Debian cloud image / netinstall, then `install.sh --non-interactive --hostname=<fqdn>`." ~15–30 lines, reuses the already-complete unattended path, serves self-hosters. This is optional and operator-discretion; it does **not** change the verdict for the issue *as written* (which is about building the pipeline).
- **Remove the `good first issue` label** (operator action) — it mis-signals difficulty and location.
- **Operator disposition:** close as re-homed, or relabel/retitle to the docs-recipe-only scope. Agents do not close issues; this is left open with the verdict label for operator visibility.

### Worth-Implementing Analysis

- **Cost:** in this repo, near-zero *correct* code — the only defensible item is the ~15–30 line docs recipe. The pipeline itself is substantial but belongs elsewhere; building it here would also carry recurring per-release retest/maintenance misfiled in a code repo.
- **Benefit:** the underlying need is real and operator-endorsed (faster clean deploys; eliminates documented stale-template clone-debt) — but that benefit is realized by the *separate tooling*, already mostly built. The marginal benefit of building it *here* is negative (duplication + contract risk).
- **Verdict rationale:** catastrophic-if-built-here is real and fleet-wide; catastrophic-if-NOT-built-here is ~nil (PMSS already serves its slice). The asymmetry is the signal.
- **Alternative:** re-home residual work to the dev-ops tooling; optionally add the docs recipe here. Reuse the post-config hook + the existing cloud-image recipe — no new abstractions.

### Merit Check — Tier A

1. **Concrete present demand.** Operator-filed and reaffirmed; demand is real but targets an image artifact + provisioning, quote: *"Companion script for proxmox install new VM with PMSS, arguments should be IP address to use, VMID."* For *this repo's code* specifically: no present demand — the install half is already shipped.
2. **Function.** When shipped it produces a sysprepped bootable PMSS disk image per Debian release plus a hypervisor-side script that runs `qm create/importdisk/set` to stand up a VM from that image given an IP and VMID. None of it executes inside the PMSS overlay.
3. **Success metric.** Within 30 days: time-to-stand-up a clean PMSS seedbox VM drops from the manual multi-step `qm` sequence to one command, and zero recurrence of stale-template clone-debt incidents — both measured on the build/hypervisor side, not in this repo.
4. **Catastrophic if implemented (here).** Orchestration added to `install.sh` → its default-invocation contract changes → unattended/break-glass installs inherit steps they cannot satisfy → installs fail on the recovery path that must stay simple → detected late, on fresh provisions, fleet-wide.
5. **Catastrophic if NOT implemented (here).** Nothing in this repo regresses — the unattended-install slice already ships and serves. The residual stays in dedicated tooling (its correct home). This near-nil downside, against the real downside in #4, is the verdict signal.
6. **Deletion alternative.** Achieve the function with zero new pipeline code here: reuse unattended `install.sh` via the separate tooling's post-config hook + the documented cloud-image/netinstall recipe.
7. **Cost.** (a) Build: ~0 correct LOC here beyond an optional docs recipe. (b) Maintenance/yr: a `qm` wrapper here would need per-Debian-release retest, `qm`-API-drift tracking, image-checksum upkeep — recurring, misfiled. (c) Blast radius: `install.sh` edits touch every install, fleet-wide. (d) Undo: docs recipe = clean revert; pipeline entangled with the install path = costly to unwind (Type-1-ish).
8. **Dependency blast (Hyrum).** `install.sh`'s default invocation is depended on by the documented one-liner installer, by unattended cron/CI, and by the break-glass path (ADR 0007). Image/provisioning consumers are separate systems (build host, hypervisor) — confirming the work's home is separate.
9. **Requirements challenge.** The dumbest requirement is "push to our static server," because it hardcodes a 2024-era channel; the real need is a versioned, fetchable image artifact, which the current pipeline already addresses.

*Tier B:* removal/weaken lens — N/A (additive). Metric/threshold — N/A. Rewrite/v2 — N/A. Performance — N/A. Irreversible (Bezos Type-1) — the risky in-repo version is one-way-door-ish (entangled with the install contract); the recommended path (don't build here / docs-only) is a clean two-way door.

### Anti-Pattern Hunt

- **REGEX_FOR_SEMANTICS:** PASS — no semantic/intent scoring via pattern-matching proposed.
- **PREMATURE_OPTIMIZATION:** PASS — the image engine Exists/Tested/Useful already (in separate tooling); rebuilding it here is the duplicate/premature path, rejected.
- **OPERATOR_HANDOFF_ANTIPATTERN:** PASS — the recommendation is architectural re-homing (work genuinely belongs in another repo), not pushing execution to the operator. The lone process smell (verdict reached, no label applied) is corrected this pass.
- **LAZY_HALT_AFTER_AUTHORIZATION:** PASS — full investigation, verdict, and label action completed; no early halt.
- **FALSE_COMPLETION_CLAIM:** PASS — no "done/fixed" claim; status is "don't build here; the PMSS slice already ships."
- **FABRICATED_AUTHORITY_OR_PATH:** PASS — every cited path verified this session (`install.sh:127-219`, `docs/adr/0007`, `docs/install.md:8,17`, `scripts/` grep, `git log -S`).
- **FAIL_OPEN_SAFETY_BEHAVIOR:** PASS — N/A to a don't-build verdict; for reference, `install.sh`'s unattended path fails safe (no controlling TTY → skip prompts, never hangs — ADR 0007).
- **DRY_RUN_THEATER:** PASS — `install.sh --dry-run` is a real parse-and-exit path, not used here as an evidence substitute.
- **INVESTIGATION_LOOP_THEATER:** PASS (conditional on the label) — #53 has cycled (2 BLOCKED build attempts + a prior deep investigation that reached a verdict but applied no label). Re-investigating again with zero state change would be loop theater; this pass instead breaks the loop by applying `should-not-implement` so the autonomous cycle stops re-selecting it.
- **LACK_OF_INVESTIGATION:** PASS — ≥2 counter-signals queried: (1) "does this repo contain any VM-templating code/history?" → `grep` + `git log -S` (both empty); (2) "does an automated Proxmox provisioning wrapper exist anywhere?" → memory + issue search (manual `qm` only).
- **DUPLICATE_CONVERGENCE_ACROSS_INSTALL_AND_UPDATE (#34):** N/A — issue does not touch `scripts/lib/update/systemPrep/*` or any idempotent boot/runtime convergence default.

### Decomposition Targets

None recommended in this code area. `install.sh` is a deliberately guarded, decade-old bootstrapper ("treat it gently"; ADR 0007) — it is explicitly exempt from the routine "split past N lines" rule, and decomposing it for this issue would be the opposite of the verdict. The only additive opportunity is the optional `docs/install.md` provisioning recipe noted above, which is documentation, not refactoring.

— Väinämöinen
